app/services/transfers/service.go

```diff
@@ -126,106 +128,132 @@ func (bs *Service) SaveRecoveredTxn(txId, amount string, asset string, m memo.Me
+		ts.logger.Errorf("Failed to get TX [%s] from DB", tsm.TransactionId)
 		return err
 	}
 
-	tx.Signature = signature
+	tx.Signature = tsm.Signature
 	tx.SignatureMsgTxId = messageTxId.String()
 	tx.Status = transaction.StatusInProgress
 	tx.SignatureMsgStatus = transaction.StatusSignatureSubmitted
-	err = bs.transactionRepository.Save(tx)
+	err = ts.transactionRepository.Save(tx)
 	if err != nil {
-		bs.logger.Errorf("Failed to update TX [%s]. Error [%s].", tm.TransactionId, err)
+		ts.logger.Errorf("Failed to update TX [%s]. Error [%s].", tsm.TransactionId, err)
 		return err
 	}
 
 	// Attach update callbacks on Signature HCS Message
-	bs.logger.Infof("Submitted signature for TX ID [%s] on Topic [%s]", tm.TransactionId, bs.topicID)
-	onSuccessfulAuthMessage, onFailedAuthMessage := bs.authMessageSubmissionCallbacks(tm.TransactionId)
-	bs.mirrorNode.WaitForTransaction(messageTxId.String(), onSuccessfulAuthMessage, onFailedAuthMessage)
+	ts.logger.Infof("Submitted signature for TX ID [%s] on Topic [%s]", tsm.TransactionId, ts.topicID)
+	onSuccessfulAuthMessage, onFailedAuthMessage := ts.authMessageSubmissionCallbacks(tsm.TransactionId)
+	ts.mirrorNode.WaitForTransaction(messageTxId.String(), onSuccessfulAuthMessage, onFailedAuthMessage)
 	return nil
 }
 
-func (bs *Service) authMessageSubmissionCallbacks(txId string) (onSuccess, onRevert func()) {
+func (ts *Service) authMessageSubmissionCallbacks(txId string) (onSuccess, onRevert func()) {
 	onSuccess = func() {
-		bs.logger.Debugf("Authorisation Signature TX successfully executed for TX [%s]", txId)
-		err := bs.transactionRepository.UpdateStatusSignatureMined(txId)
+		ts.logger.Debugf("Authorisation Signature TX successfully executed for TX [%s]", txId)
+		err := ts.transactionRepository.UpdateStatusSignatureMined(txId)
 		if err != nil {
-			bs.logger.Errorf("Failed to update status for TX [%s]. Error [%s].", txId, err)
+			ts.logger.Errorf("Failed to update status for TX [%s]. Error [%s].", txId, err)
 			return
 		}
 	}
 
 	onRevert = func() {
-		bs.logger.Debugf("Authorisation Signature TX failed for TX ID [%s]", txId)
-		err := bs.transactionRepository.UpdateStatusSignatureFailed(txId)
+		ts.logger.Debugf("Authorisation Signature TX failed for TX ID [%s]", txId)
+		err := ts.transactionRepository.UpdateStatusSignatureFailed(txId)
 		if err != nil {
-			bs.logger.Errorf("Failed to update status for TX [%s]. Error [%s].", txId, err)
+			ts.logger.Errorf("Failed to update status for TX [%s]. Error [%s].", txId, err)
 			return
 		}
 	}
 	return onSuccess, onRevert
 }
+
+func (ts *Service) SignAuthorizationMessage(txId, ethAddress, erc20Address, amount, fee, gasPriceWei string) (string, error) {
+	authMsgHash, err := auth_message.EncodeBytesFrom(txId, ethAddress, erc20Address, amount, fee, gasPriceWei)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Failed to encode the authorisation signature for TX ID [%s]. Error: %s", txId, err))
```

Comment:
This should be a log message imo. Errors are normally short, concise and as it turns out all lower case in Go.
I would suggest to log here the error and return `err`. The place you are calling this should only state that this exact operation failed without logging the error. That is how we are doing it in the other `services` functions.

There are 2 places from which  `SignAuthorizationMessage` is called (this service and the recovery process). Once you reformat the `ProcessTranfer` function and move the logic to the `Handler`, this function will be called only from the 2 processes -> `handler` and `recovery`